src/test/optimizer/strategy/join_predicate_ordering_rule_test.cpp

```diff
@@ -0,0 +1,123 @@
+    a_z = node_a->get_column("z");
+
+    node_b = MockNode::make(
+        MockNode::ColumnDefinitions{{DataType::Int, "x"}, {DataType::Int, "y"}, {DataType::Int, "z"}}, "b");
+    b_x = node_b->get_column("x");
+    b_y = node_b->get_column("y");
+    b_z = node_b->get_column("z");
+  }
+
+  std::shared_ptr<JoinPredicateOrderingRule> _rule;
+  std::shared_ptr<MockNode> node_a, node_b;
+  LQPColumnReference a_x, a_y, a_z, b_x, b_y, b_z;
+};
+
+TEST_F(JoinPredicateOrderingRuleTest, InnerEquiJoin) {
+  set_statistics_for_mock_node(node_b, 100,
+                               {GenericHistogram<int32_t>::with_single_bin(0, 40, 100, 5),
+                                GenericHistogram<int32_t>::with_single_bin(30, 70, 100, 5),
+                                GenericHistogram<int32_t>::with_single_bin(10, 50, 100, 5)});
+
+  // Assuming equals_(a_y, b_y) < equals_(a_z, b_z) < equals_(a_x, b_x) wrt. estimated cardinality. This might
+  //  change when our cardinality estimator changes. In that case the histograms should be updated.
+  const auto expected_join_predicates = expression_vector(equals_(a_y, b_y), equals_(a_z, b_z), equals_(a_x, b_x));
+  const auto expected_lqp = JoinNode::make(JoinMode::Inner, expected_join_predicates, node_a, node_b);
+
+  for (const auto& input_join_predicates :
+       {expression_vector(equals_(a_y, b_y), equals_(a_z, b_z), equals_(a_x, b_x)),
+        expression_vector(equals_(a_x, b_x), equals_(a_y, b_y), equals_(a_z, b_z)),
+        expression_vector(equals_(a_y, b_y), equals_(a_x, b_x), equals_(a_z, b_z))}) {
+    SCOPED_TRACE(std::accumulate(input_join_predicates.begin(), input_join_predicates.end(),
+                                 std::string{"input predicates: "},
+                                 [](const auto& predicates_string, const auto& predicate) {
+                                   return predicates_string + '[' + predicate->as_column_name() + "] ";
+                                 }));
+
+    const auto input_lqp = JoinNode::make(JoinMode::Inner, input_join_predicates, node_a, node_b);
+
+    const auto actual_lqp = StrategyBaseTest::apply_rule(_rule, input_lqp);
+    EXPECT_LQP_EQ(actual_lqp, expected_lqp);
+  }
+}
+
+TEST_F(JoinPredicateOrderingRuleTest, AntiNonEqualsJoin) {
+  set_statistics_for_mock_node(node_b, 100,
+                               {GenericHistogram<int32_t>::with_single_bin(0, 40, 100, 5),
+                                GenericHistogram<int32_t>::with_single_bin(30, 70, 100, 5),
+                                GenericHistogram<int32_t>::with_single_bin(10, 50, 100, 5)});
+
+  const auto non_equals_predicates = expression_vector(greater_than_(a_y, b_y), less_than_(a_z, b_z));
+
+  for (const auto& join_mode : {JoinMode::Inner, JoinMode::Left, JoinMode::Right, JoinMode::FullOuter}) {
+    const auto input_lqp = JoinNode::make(join_mode, non_equals_predicates, node_a, node_b);
+    EXPECT_NO_THROW(StrategyBaseTest::apply_rule(_rule, input_lqp));
```

Comment:
test output data for input data + operation. Testing that something doesn't throw is not a test.

Reply:
Well it's still making sure the rule is not throwing for all JoinModes. At the moment greater_than_ and less_than_ predicates are not estimated, so I can't test the output. I'd just drop a TODO(anyone) to add an actual test as soon as greater_than_ and less_than_ predicates can be estimated.

Reply:
google test itself will report an exception, so EXPECT_NO_THROW is in itself pretty non sensical. As long as there is any deterministic output (is there) it should be tested. If there really really is no deterministic output, a EXPECT_NO_THROW with a comment is a last resort.

Reply:
Alright I'm testing the  output now.